packages/web-app-files/src/store/mutations.js

```diff
@@ -329,6 +340,31 @@ export default {
     $_upsertResource(state, resource, false)
   },
 
+  /**
+   * Updates a single resource key. If the resource with given id doesn't exist nothing will happen.
+   *
+   * @param state Current state of this store module
+   * @param params.id Id of the resource to be updated
+   * @param params.key the resource key that the value should be applied to
+   * @param params.value the value that will be attached to the key
+   */
+  UPDATE_RESOURCE_KEY(state, params) {
```

Comment:
Can't we just replace `UPDATE_RESOURCE` entirely? i.e. use this mutation body as the body of `UPDATE_RESOURCE` and then move the code from the `$_upsertResource` helper into `UPSERT_RESOURCE`?

Reply:
`UPDATE_RESOURCE_KEY` sounds misleading. Like it would update the key, while it really updates the resource.

Reply:
Sorry for the confusion. I had something wrong in mind. Changing the mutation name to `UPDATE_RESOURCE_FIELD` is sufficient.